### ISSUE TITLE
global: fix PEP8 issues

### ISSUE THREAD

**Issue by jirikuncar** [inveniosoftware/invenio-workflows#2]

**Comment by jalavik** [comment]
Please for the love of all that is holy, do not do this until PR https://github.com/inveniosoftware/invenio/pull/2628 and https://github.com/inveniosoftware/invenio/pull/3136 have been ported to this module.  @dset0x

**Comment by tiborsimko** [comment]
@jalavik :+1: :smile:

**Comment by jalavik** [comment]
I'd suggest to add E501 (line too long) to the ignore list of pytest.ini. By adding it, I only found minimal PEP8 issues remaining. This would allow us to solve this issue while still remaining sane.

To quote Guido
> But most importantly: know when to be inconsistent -- sometimes the style guide just doesn't apply. When in doubt, use your best judgment.

And a point often observed when mindlessly trying to adhere to E501:
> When applying the guideline would make the code less readable, even for someone who is used to reading code that follows this PEP.

https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds

@tiborsimko @jirikuncar

**Comment by jirikuncar** [comment]
@jalavik if you think that it make readability worse then disable it per line/block bases using e.g. `# noqa`.

**Comment by jalavik** [comment]
@jirikuncar Right. We could do that. But this is not the main issue here. I agree we can enable the PEP8 stuff, but please do it after we have merged the code that is already developed. It's unnecessarily painful to cause extra work when porting the existing PRs. We have more important things to do honestly.

**Comment by tiborsimko** [comment]
I'd say (1) let's complete functionality and review and integrate all the pre-existing PRs regardless of any style issues they may have, and (2) let's amend PEP-8 compliance afterwards. The former is indeed quite more important than the latter -- but also this does not mean that the latter should be disregarded! We collectively decided in the past to go for PEP-8 compliance in Invenio projects, which ideally includes E501 as well.  (The advantages include a possibility to quickly jump around Python projects/packages and always feeling at home.)  To me, the ideal would be to take on the complete PEP-8 habits from the get-go, e.g. I'd recommend everyone to switch on [full flake8 checking in their editors](https://github.com/inveniosoftware/invenio/issues/2482#issuecomment-61113830) so that PRs are already PEP-8 compliant by inherent nature, without any extra work or extra thinking.

In summary, while I fully agree that cleaning E501 for big pre-existing branches can wait, for all that love that @jalavik mentioned :smile: I'd also say let's not loose PEP-8 from sight either!  It's not that demanding to comply if one has the checker right at the fingertips when developing... And as for the ideal line length and code readability, I'd say let's stick with the defaults as much as possible... People reading code on their mobile phones, for example, would appreciate our E501 compliance :wink:

**Comment by jalavik** [comment]
> but also this does not mean that the latter should be disregarded! ... To me, the ideal would be to take on the complete PEP-8 habits from the get-go

Fully agree. :+1: With this issue moved to `0.2.0` milestone I think we are good in this regard. We can apply the pre-existing PRs and then amend PEP8 (including E501) before releasing `0.2.0`.

**Comment by jirikuncar** [comment]
@jalavik do you have any pending branches for 0.2.0? Can you create at least WIP PR? Thanks

**Comment by jalavik** [comment]
@jirikuncar Yeah, see #8

**Comment by jalavik** [comment]
Fixed with new Invenio 3 branch. See #25
